design/carvel-install-design.md

```diff
@@ -0,0 +1,24 @@
+# Design proposal for Velero Installation with Carvel
+
+Currently, the Velero CLI tool has a `install` command that configures numerous major and minor aspects of Velero. This document outlines how we can also have Velero be installed and upgraded using the Carvel tooling.
+
+
+## Goals
+- Have one source of truth for all Velero configuration
+- Have compatibility with gitop practices (i.e. ability to generate a full set of yaml for install that can be stored in source control)
+- Have a clear path for deprecating of the `install` command
+
+## Requirements
+- Ability for the Carvel deployment to consume/inject configuration data as needed for a normal Velero deployment with default resources. For example: it should configure a default BSL alongside the deployment.
+- The Carvel toolset must be able to, in addition to install, upgrade Velero as well as update its configuration settings.
+- "Ask once" approach to configuring the Velero deployment defaults. In other words: reuse given parameters for putting together the needed overlays and packaging configuration.
+- Ideally, setup the Carvel tooling so that it discovers and uses configuration data from known distros.
+- The toolset and artifacts must be setup in a way that it will access and consume the CRDs that match the intended Velero version. Users should have uncomplicated access to the proper versions of the CRDs to be able to upgrade and downgrade Velero.
+- The toolset and artifacts must be laid out in a directory structure in a way that it can be discovered and used by other projects.
+- Migrate deployment configuration from using Go code to using Carvel manifests, and wrap the `install` command around the toolset.
+- Deprecate the `install` command once the Carvel tooling is fully setup and well tested.
```

Comment:
I don't think we'll actually be able to deprecate "install", since Carvel also relies on installing KAPP which seems like a lot to ask the non-Tanzu user to do to install Velero.  I think it would be possible, though, to call the Karvel libraries from the CLI and have the CLI use Carvel as a library to generate the installation yamls

Reply:
I'm removing this line because now that I think about it, that's what the line above says (wrap aroun a Carvel library).

Reply:
Oh lord. Please ignore my ramblings above, my mind was elsewhere. Apologies!

@dsu-igeek  I was very much assuming that we could require the installation and use of any of the Carvel tooling for the non Tanzu Velero user. As far as I am understanding, this would mean ytt for configuration and kapp for deployment for the requirements we have. Is this not correct? We can require ytt but not kapp? Something else?

Reply:
Explanation: I read your KAPP in all caps and my brain had translated it into kapp-controller, but my comment doesn't make sense even so. Ha.